daphne/src/messages.rs

```diff
@@ -809,6 +817,218 @@ impl Decode for HpkeCiphertext {
     }
 }
 
+#[derive(Clone, Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum VdafType {
+    Prio3AES128Count,
+    Prio3AES128Sum { bit_length: u8 },
+    Prio3AES128Histogram { buckets: Vec<u64> },
+    Poplar1AES128 { bit_length: u16 },
+    NotImplemented(u32),
```

Comment:
Typically we would want to store the payload as well, like we do for Extension::Unhandled. However, this makes me realize there is a problem with the spec: How does the decoder know how long the rest of VdafConfig is supposed to be if it gets an unrecognized VdafType? This seems like a bug in the spec. (I'll raise an issue.)

The fix would be to length-prefix the remainder of the VdafConfig message. This would be a breaking change, so it'll have to wait for taskprov-01; I think the only way to handle this situation in taskprov-00 is to hard-fail if we don't recognized the VdafType.

Reply:
It's ok at the moment as it is last, but it is a problem if the task config is ever changed in the future.  Though you could argue if you were going to change it, then you should allocate a different extension code anyway.  I'm not objecting to clarifying this part of the spec though!

Reply:
> Though you could argue if you were going to change it, then you should allocate a different extension code anyway.

that sounds right, but we're expecting there to be lots of VDAFs in the wild for which there is no formal specification. At least, for the foreseeable future.